src/JsonApiDotNetCore.MongoDb/MongoEntityRepository.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using JsonApiDotNetCore.Configuration;
+using JsonApiDotNetCore.MongoDb.Extensions;
+using JsonApiDotNetCore.Queries.Internal.QueryableBuilding;
+using JsonApiDotNetCore.Repositories;
+using JsonApiDotNetCore.Resources;
+using MongoDB.Driver;
+using MongoDB.Driver.Linq;
+using JsonApiDotNetCore.Queries;
+using JsonApiDotNetCore.Queries.Expressions;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+
+namespace JsonApiDotNetCore.MongoDb
+{
+    public class MongoEntityRepository<TResource, TId>
+        : IResourceRepository<TResource, TId>
+        where TResource : class, IIdentifiable<TId>
+    {
+        private readonly IMongoDatabase _db;
+        private readonly ITargetedFields _targetedFields;
+        private readonly IResourceContextProvider _resourceContextProvider;
+        private readonly IResourceFactory _resourceFactory;
+        
+        public MongoEntityRepository(
+            IMongoDatabase db,
+            ITargetedFields targetedFields,
+            IResourceContextProvider resourceContextProvider,
+            IResourceFactory resourceFactory)
+        {
+            _db = db;
+            _targetedFields = targetedFields;
+            _resourceContextProvider = resourceContextProvider;
+            _resourceFactory = resourceFactory;
+        }
+
+        private IMongoCollection<TResource> Collection => _db.GetCollection<TResource>(typeof(TResource).Name);
+        private IMongoQueryable<TResource> Entities => Collection.AsQueryable();
```

Comment:
Instead of this, a protected virtual method `GetAll()` should be provided (and used from all places), so that users can override and put custom logic in there. For example, to add an extra filter condition for authorization.